gtda/externals/bindings/subsampling_bindings.cpp

```diff
@@ -0,0 +1,46 @@
+/******************************************************************************
+ * Description:      gudhi's subsampling interfacing with pybind11
+ * License:          GNU AGPLv3
+ *****************************************************************************/
+
+#include <Subsampling_interface.h>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace py = pybind11;
+
+PYBIND11_MODULE(gtda_subsampling, m) {
+  using namespace pybind11::literals;
+
+  m.def("subsampling_n_farthest_points",
+        py::overload_cast<const std::vector<std::vector<double>>&, unsigned>(
+            &Gudhi::subsampling::subsampling_n_farthest_points));
+  m.def("subsampling_n_farthest_points",
+        py::overload_cast<const std::vector<std::vector<double>>&, unsigned,
+                          unsigned>(
+            &Gudhi::subsampling::subsampling_n_farthest_points));
+
+  m.def("subsampling_n_farthest_points_from_file",
+        py::overload_cast<const std::string&, unsigned>(
+            &Gudhi::subsampling::subsampling_n_farthest_points_from_file));
+  m.def("subsampling_n_farthest_points_from_file",
+        py::overload_cast<const std::string&, unsigned, unsigned>(
+            &Gudhi::subsampling::subsampling_n_farthest_points_from_file));
+
+  m.def("subsampling_n_random_points",
+        py::overload_cast<const std::vector<std::vector<double>>&, unsigned>(
+            &Gudhi::subsampling::subsampling_n_random_points));
+  m.def("subsampling_n_random_points_from_file",
+        py::overload_cast<const std::string&, unsigned>(
+            &Gudhi::subsampling::subsampling_n_random_points_from_file));
+
+  m.def("subsampling_sparsify_points",
+        py::overload_cast<const std::vector<std::vector<double>>&, double>(
+            &Gudhi::subsampling::subsampling_sparsify_points));
+  m.def("subsampling_sparsify_points_from_file",
+        py::overload_cast<const std::string&, double>(
+            &Gudhi::subsampling::subsampling_sparsify_points_from_file));
```

Comment:
Now that I have more experience with `pybind11`, I try to add two more information to the `m.def(...)`:

* How to give [arguments names](https://pybind11.readthedocs.io/en/stable/basics.html#keyword-arguments), I have a preference for the shorthand notation `"<variable_name>"_a`, you can also specify default values if it's necessary
* Hot to [document functions](https://pybind11.readthedocs.io/en/stable/basics.html#creating-bindings-for-a-simple-function), last argument of `m.def` can be a short description of the function, add it if the name of the function is not clear enough.

Reply:
Please go ahead, you can make a PR to my fork's branch

Reply:
Done, I pushed on your branch directly if you don't mind